deploy/helm/templates/deployment.yaml

```diff
@@ -0,0 +1,94 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "starboard-operator.fullname" . }}
+  labels:
+    {{- include "starboard-operator.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "starboard-operator.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- . | toYaml | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "starboard-operator.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: "{{ .Chart.Name }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          env:
+            - name: OPERATOR_NAMESPACE
+              value: {{ .Release.Namespace | quote }}
+            - OPERATOR_TARGET_NAMESPACES:
+              value: {{ .Values.targetNamespaces | quote }}
+            - name: OPERATOR_METRICS_BIND_ADDRESS
+              value: ":8080"
+            - name: OPERATOR_HEALTH_PROBE_BIND_ADDRESS
+              value: ":9090"
+            {{- /* Import configuration from passed Helm values */}}
+            {{- range $key, $value := .Values.envSecret.stringData }}
+            {{- if $value }}
+            - name: {{ $key | quote }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.envSecret.name | default (include "starboard-operator.fullname" $) | quote }}
+                  key: {{ $key | quote }}
+                  optional: true
+            {{- end }}
+            {{- end }}
```

Comment:
I made a choice here to ensure that we avoid future maintenance trouble, by letting the configuration be set from passed values without needing to match specific hardcoded values.